features/net/network-socket/nsapi_dns.cpp

```diff
@@ -88,7 +89,7 @@ static void dns_append_question(uint8_t **p, const char *host)
     dns_append_word(p, 1); // qclass = 1
 }
 
-static bool dns_scan_response(const uint8_t **p, nsapi_addr_t *addr)
+static int dns_scan_response(const uint8_t **p, nsapi_addr_t *addr, unsigned addr_count)
```

Comment:
Sorry, missed this comment the first time. This function is simply returning error codes.

Like the BSD socket api, the network-socket's error codes are in a small range that is garunteed to fit in the minimum bound of an int on a standard C compiler (16-bits, [wiki](https://en.wikipedia.org/wiki/C_data_types)). If you are not on a standard C compiler, there are going to be a lot of other problems.

The correct type would actually be `int_fast16_t`, but this actually has no semantic benefit over `int` short of being more explicit.

I would actually vote for separate nsapi types, (ie `nsapi_size_t`, `nsapi_error_t`, `nsapi_size_or_error_t`), although I haven't had the chance to propose this patch yet.